elements/lisk-p2p/src/p2p.ts

```diff
@@ -823,12 +823,13 @@ export class P2P extends EventEmitter {
 						DUPLICATE_CONNECTION,
 						DUPLICATE_CONNECTION_REASON,
 					);
-				} else {
-					this._peerPool.addInboundPeer(incomingPeerInfo, socket);
-					this.emit(EVENT_NEW_INBOUND_PEER, incomingPeerInfo);
+
+					return;
 				}
 
 				try {
+					this._peerPool.addInboundPeer(incomingPeerInfo, socket);
+					this.emit(EVENT_NEW_INBOUND_PEER, incomingPeerInfo);
 					this._peerBook.addPeer(incomingPeerInfo);
 				} catch (error) {
```

Comment:
Shall we use the same logic of throwing an error when adding a peer to the pool as we do in the peer book?

Reply:
what do you mean by using the same logic as peer book?
I think this logic shouldn't throw error on the ExistingPeerError, which might happen on the peer book, and if anything else happen, the application will throw unhandle error?

Reply:
We can make it consistent with `addPeer` approach of peerbook.
we don't have to check for `const existingPeer = this._peerPool.getPeer(peerId);` if handle it separately.

```typescript
try {
	this._peerPool.addInboundPeer(incomingPeerInfo, socket);
	this.emit(EVENT_NEW_INBOUND_PEER, incomingPeerInfo);
} catch (err) {
         this._disconnectSocketDueToFailedHandshake(
         socket,
	 DUPLICATE_CONNECTION,
	 DUPLICATE_CONNECTION_REASON);

        return;
}

try {
	this._peerBook.addPeer(incomingPeerInfo);
} catch (error) {
        if (!(error instanceof ExistingPeerError)) {
	     throw error;
	}
}
```